src/documents/views/__tests__/NewDocumentView.test.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable no-loop-func */
+import React from 'react';
+import wait from 'waait';
+import {MockedProvider} from '@apollo/react-testing';
+import {MemoryRouter} from 'react-router-dom';
+import {render, act, fireEvent, cleanup} from 'react-testing-library';
+import {mocks} from '../../../../__mocks__/kf-api-study-creator/mocks';
+import NewDocumentView from '../NewDocumentView';
+
+afterEach(cleanup);
+
+let file = {
+  lastModified: 1567797130510,
+  name: 'organize.tsv',
+  size: 7,
+  type: 'text/tab-separated-values',
+  webkitRelativePath: '',
+};
+
+const historyMock = {
+  push: jest.fn(),
+  location: {pathname: '/study/SD_8WX8QQ06/documents/new-documents'},
+};
+
+it('Render out the new document view', async () => {
+  const tree = render(
+    <MockedProvider mocks={mocks}>
+      <MemoryRouter
+        initialEntries={['/study/SD_8WX8QQ06/documents/new-document']}
+      >
+        <NewDocumentView
+          match={{params: {kfId: 'SD_8WX8QQ06'}}}
+          location={{state: {file: file}}}
+          history={historyMock}
+        />
+      </MemoryRouter>
+    </MockedProvider>,
+  );
+
+  await wait();
+
+  expect(tree.container).toMatchSnapshot();
```

Comment:
why take snapshots if we're not asserting that they should look a particular way?

Reply:
This one is the look when a file is just uploaded, user hasn't type anything in the file name yet. The next snapshot test is user inputing. This will be different comparing to the snapshot at line 98, no warning when the fields haven't been touched vs. show warning when the name and description are touched but input is empty.